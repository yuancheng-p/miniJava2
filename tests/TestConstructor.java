package tutu.titi.toto;

//class A can pass the test
public class A {
    int a1 = 1;
    int b1 = 2;
    public A(int j,int k, int p){
      a1 = j;
      b1 = k;
      int i = 999;
    }
    public static void main(String [] args) {
      A a = toto9.new A(4,5,9);
    }

}
