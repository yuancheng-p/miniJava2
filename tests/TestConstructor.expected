successfull parsing
package tutu.titi.toto
public class A extends Object {
  int a1 = 1;
  int b1 = 2;
  A(int j,int k,int p) {
    a1=j;
    b1=k;
    int i 999;
  }
  void main(String[] args)  {
    A a toto9.new A(4,5,9);
  }
}

start typing
check add and methods:A:main
--------- env --------
A : class_env(parent=: Object)
method : main paramlist: String[] 
 attr:  int b1 = 2;
 attr:  int a1 = 1;
Boolean : class_env(parent=: Object)
Integer : class_env(parent=: Object)
Object : class_env(parent=: )
String : class_env(parent=: Object)
end typing
main
######################### Call Constructor: A
+++Assign:4 (int)
------ current frame ------
k : 5 (int)
p : 9 (int)
j : 4 (int)
this : 0 (ref)
=== heap ===
[A] (0) :
  b1 : 2 (int)
  a1 : 4 (int)
+++Assign:5 (int)
------ current frame ------
k : 5 (int)
p : 9 (int)
j : 4 (int)
this : 0 (ref)
=== heap ===
[A] (0) :
  b1 : 5 (int)
  a1 : 4 (int)
------ current frame ------
k : 5 (int)
p : 9 (int)
j : 4 (int)
this : 0 (ref)
i : 999 (int)
=== heap ===
[A] (0) :
  b1 : 5 (int)
  a1 : 4 (int)
######################### Finished Call Constructor: A
------ current frame ------
a : 0 (ref)
=== heap ===
[A] (0) :
  b1 : 5 (int)
  a1 : 4 (int)
